lj/lj.go

```diff
@@ -18,17 +18,24 @@
 // Package lj implements common lumberjack types and functions.
 package lj
 
+import "net"
+
 // Batch is an ACK-able batch of events as has been received by lumberjack
 // server implemenentations. Batches must be ACKed, for the server
 // implementations returning an ACK to it's clients.
 type Batch struct {
 	Events []interface{}
+	Conn   net.Conn
```

Comment:
I don't like that this exposes the underlying `net.Conn` to the library users. They should not be able to directly read/write/close the underlying connection.

I think I would want to only expose the remote `net.Addr` and a `*tls.ConnectionState` if TLS was used.